exercises/strings/replaceCharacter/replaceCharacter.js

```diff
@@ -13,11 +13,32 @@
  */
 
 function replaceCharacter(string, target, replaceWith) {
+  newWord = ""
+  startindex = 0
+  let targetindex = []
+  for (i in string){
+    if (string[i] === target){
+      targetindex.push(i)
+    }
+  }
+  for (i of targetindex) {
+    newWord = newWord.slice(startindex, i) //+ replaceWith
```

Comment:
Every time you call `slice` it creates a new string. You may as well append to `newWord` as you go, as in:

```js
function replaceCharacter(string, target, replaceWith) {
  let result = '';

  for (let char of string) {
    if (char === target) {
      result += target;
    } else {
      result += char;
    }
  }

  return result;
}
```